AppInspector.CLI/Writers/WriterFactory.cs

```diff
@@ -0,0 +1,250 @@
+                default:
+                    WriteOnce.Error(MsgHelp.FormatString(MsgHelp.ID.CMD_INVALID_ARG_VALUE, "-f"));
+                    throw new OpException((MsgHelp.FormatString(MsgHelp.ID.CMD_INVALID_ARG_VALUE, "-f")));
+            }
+
+            //assign the stream as a file or console
+            writer.OutputFileName = cLIAnalyzeCmdOptions.OutputFilePath;
+            writer.TextWriter = GetTextWriter(writer.OutputFileName);
+
+            return writer;
+
+        }
+
+
+        static CommandResultsWriter GetTagTestWriter(CLICommandOptions options)
+        {
+            CLITagTestCmdOptions cLIAnalyzeCmdOptions = (CLITagTestCmdOptions)options;
+            CommandResultsWriter writer = null;
+
+            switch (cLIAnalyzeCmdOptions.OutputFileFormat.ToLower())
+            {
+                case "_dummy":
+                    writer = new TagTestDummyWriter();
+                    break;
+                case "json":
+                    writer = new TagTestJsonWriter();
+                    break;
+                case "text":
+                    writer = new TagTestTextWriter();
+                    break;
+                default:
+                    WriteOnce.Error(MsgHelp.FormatString(MsgHelp.ID.CMD_INVALID_ARG_VALUE, "-f"));
+                    throw new OpException((MsgHelp.FormatString(MsgHelp.ID.CMD_INVALID_ARG_VALUE, "-f")));
+            }
+
+            //assign the stream as a file or console
+            writer.OutputFileName = cLIAnalyzeCmdOptions.OutputFilePath;
+            writer.TextWriter = GetTextWriter(writer.OutputFileName);
+
+            return writer;
+        }
+
+        static CommandResultsWriter GetTagDiffWriter(CLICommandOptions options)
+        {
+            CLITagDiffCmdOptions cLIAnalyzeCmdOptions = (CLITagDiffCmdOptions)options;
+            CommandResultsWriter writer = null;
+
+            switch (cLIAnalyzeCmdOptions.OutputFileFormat.ToLower())
+            {
+                case "_dummy":
+                    writer = new TagDiffDummyWriter();
+                    break;
+                case "json":
+                    writer = new TagDiffJsonWriter();
+                    break;
+                case "text":
+                    writer = new TagDiffTextWriter();
+                    break;
+                default:
+                    WriteOnce.Error(MsgHelp.FormatString(MsgHelp.ID.CMD_INVALID_ARG_VALUE, "-f"));
+                    throw new OpException((MsgHelp.FormatString(MsgHelp.ID.CMD_INVALID_ARG_VALUE, "-f")));
+            }
+
+            //assign the stream as a file or console
+            writer.OutputFileName = cLIAnalyzeCmdOptions.OutputFilePath;
+            writer.TextWriter = GetTextWriter(writer.OutputFileName);
+
+            return writer;
+        }
+
+
+        static CommandResultsWriter GetVerifyRulesWriter(CLICommandOptions options)
+        {
+            CLIVerifyRulesCmdOptions cLIAnalyzeCmdOptions = (CLIVerifyRulesCmdOptions)options;
```

Comment:
If the first thing you do is cast this should be part of the method declaration. Otherwise this could throw an error if you give it the wrong type of command options.

Reply:
Agreed.  Fixed.